test/integration/sign-test.ts

```diff
@@ -20,12 +20,22 @@ test("sign(secret) without eventPayload throws", () => {
   expect(() => sign.bind(null, secret)()).toThrow();
 });
 
-test("sign(secret, eventPayload) with eventPayload as object returns expected signature", () => {
+test("sign(secret, eventPayload) with eventPayload as object returns expected sha1 signature", () => {
   const signature = sign(secret, eventPayload);
   expect(signature).toBe("sha1=d03207e4b030cf234e3447bac4d93add4c6643d8");
 });
 
-test("sign(secret, eventPayload) with eventPayload as string returns expected signature", () => {
+test("sign(secret, eventPayload) with eventPayload as object returns expected sha256 signature", () => {
+  const signature = sign(secret, eventPayload);
+  expect(signature).toBe("sha256=kljsdfjkldasjfkaejkfwekuweur3298r2398r89fdsuf3");
+});
+
+test("sign(secret, eventPayload) with eventPayload as string returns expected sha1 signature", () => {
   const signature = sign(secret, JSON.stringify(eventPayload));
   expect(signature).toBe("sha1=d03207e4b030cf234e3447bac4d93add4c6643d8");
 });
+
+test("sign(secret, eventPayload) with eventPayload as string returns expected sha256 signature", () => {
+  const signature = sign(secret, JSON.stringify(eventPayload));
+  expect(signature).toBe("sha256=kljsdfjkldasjfkaejkfwekuweur3298r2398r89fdsuf3");
```

Comment:
In order to support this I only came up with a modification with the signature with an optional third parameter with Webhook header's, which will be the one determining if is going to use `sha1` or `sha256`. Would that be the approach to follow or you had something different in mind?

```typescript
function sign(secret: string, payload: string | object, headers?: string): string {
  // @ts-ignore throw friendly error message when required options are missing
  if (!secret || !payload) {
    throw new TypeError("secret & payload required");
  }

  payload =
    typeof payload === "string" ? payload : toNormalizedJsonString(payload);
  
  const algorithm = containsSha256Header(headers)? "sha256" : "sha1"
  return `${algorithm}=${createHmac(algorithm, secret).update(payload).digest("hex")}`;
}
```

Reply:
I see. I see why the additional parameter is needed. I think passing the `headers` doesn't feel right here, the `sign` method does not need to be aware of all headers, it only needs to know which algorithm to use. 

How about this:

1. By default `sign(secret, payload)` returns a signature based on the `sha1` algorithm, so we don't introduce a breaking change. We  can change the default with the next breaking version.
2. In order to get a a signature using the `sha-256` algorithm, we can either introduce a third argument `algorithm?: string`, or we allow the first argument to be an object with two keys: `sign({ secret, algorithm }, payload)`. I think I'd prefer the latter. 

What do you think @dominguezcelada @wolfy1339?

Reply:
Option 2 would be okay

Reply:
I prefer option 2, so we make options object more extensible instead of a long list of arguments for the method. I go for it